crates/sui-core/src/consensus_handler.rs

```diff
@@ -332,21 +332,14 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
                         .observe(serialized_transaction.len() as f64);
                     if matches!(
                         &transaction.kind,
-                        ConsensusTransactionKind::UserTransaction(_)
+                        ConsensusTransactionKind::CertifiedTransaction(_)
                     ) {
                         self.last_consensus_stats
                             .stats
                             .inc_num_user_transactions(authority_index as usize);
                     }
-                    if let ConsensusTransactionKind::RandomnessStateUpdate(randomness_round, _) =
-                        &transaction.kind
-                    {
-                        // These are deprecated and we should never see them. Log an error and eat the tx if one appears.
-                        error!("BUG: saw deprecated RandomnessStateUpdate tx for commit round {round:?}, randomness round {randomness_round:?}")
-                    } else {
-                        let transaction = SequencedConsensusTransactionKind::External(transaction);
```

Comment:
I think it is safe to remove the RandomnessStateUpdate variant. IIUC it is not used in meaningful logic for at least a release now. And ConsensusTransaction is not persisted across epochs. There doesn't seem to be compatibility issues in tests. But let me know if you think otherwise and I can revert the commit to remove RandomnessStateUpdate.

Reply:
I agree it's safe to remove the variant. I am asking about the two lines in the "else" branch here.

Reply:
I see now that they are not removed, maybe it changed or maybe I was just reading the diff wrong on github - sorry!